worlds/earthbound/__init__.py

```diff
@@ -0,0 +1,601 @@
+import os
+import typing
+import threading
+import pkgutil
+from typing import List, Set, Dict, TextIO, Tuple
+
+from BaseClasses import Item, MultiWorld, Location, Tutorial, ItemClassification
+from Fill import fill_restrictive
+from worlds.AutoWorld import World, WebWorld
+import itertools
+import settings
+from .Items import get_item_names_per_category, item_table
+from .Locations import get_locations
+from .Regions import init_areas, connect_area_exits
+from .Options import EBOptions, eb_option_groups
+from .setup_game import setup_gamevars, place_static_items
+from .modules.enemy_data import initialize_enemies
+from .modules.flavor_data import create_flavors
+from .game_data.local_data import item_id_table, world_version
+from .modules.hint_data import setup_hints
+from .game_data.text_data import spoiler_psi, spoiler_starts, spoiler_badges
+from .Client import EarthBoundClient
+from .Rules import set_location_rules
+from .Rom import patch_rom, EBProcPatch, valid_hashes
+from .game_data.static_location_data import location_ids, location_groups
+from .modules.equipamizer import EBArmor, EBWeapon
+from .modules.boss_shuffle import BossData, SlotInfo
+from worlds.generic.Rules import add_item_rule
+from Options import OptionError
+
+
+class EBSettings(settings.Group):
+    class RomFile(settings.SNESRomPath):
+        """File name of the EarthBound US ROM"""
+        description = "EarthBound ROM File"
+        copy_to = "EarthBound.sfc"
+        md5s = valid_hashes
+
+    rom_file: RomFile = RomFile(RomFile.copy_to)
+
+
+class EBWeb(WebWorld):
+    theme = "ocean"
+
+    setup_en = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the EarthBound randomizer"
+        "and connecting to an Archipelago server.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Pink Switch"]
+    )
+
+    tutorials = [setup_en]
+
+    option_groups = eb_option_groups
+    # option_presets = eb_option_presets
+
+
+class EarthBoundWorld(World):
+    """EarthBound is a contemporary-themed JRPG. Take four psychically-endowed children
+       across the world in search of 8 Melodies to defeat Giygas, the cosmic evil."""
+    
+    game = "EarthBound"
+    option_definitions = EBOptions
+    data_version = 1
+    required_client_version = (0, 5, 0) 
+
+    item_name_to_id = {item: item_table[item].code for item in item_table if item_table[item].code}
```

Comment:
```suggestion
    item_name_to_id = {item: data.code for item, data in item_table.items() if data.code}
```
Optional but I've found this appeases the type checkers for the same result